src/ShaderGen/Hlsl/HlslBackend.cs

```diff
@@ -196,63 +178,43 @@ StructureDefinition[] orderedStructures
 
                 foreach (ResourceDefinition rd in set)
                 {
+                    if (!resourcesUsed.Contains(rd))
```

Comment:
I think this is the problem - it’s necessary to increment the resource index, even if the resource isn’t used.

Reply:
On a related note, I’ve been thinking it would be nice to test against known-good reference outputs. But I wanted to do some PRs to clean up the output formatting before suggesting that.

Reply:
> it’s necessary to increment the resource index, even if the resource isn’t used.

You're right, that is broken. @feliwir Do you have any time to submit a fix for that?

Reply:
Will do in 2hrs when home from work